.github/workflows/continuous-integration.yml

```
@@ -201,6 +202,69 @@ jobs:
         id: get-broken-link-info
         run: node ./script/github-actions/check-broken-links.js ${{ matrix.buildtype }}
 
-      - name: Slack notification
-        run: echo Slack Message ${{ steps.get-broken-link-info.outputs.SLACK_MESSAGE }}. Outputting color for testing ${{ steps.get-broken-link-info.outputs.SLACK_COLOR }}
-        if: ${{ (steps.get-broken-link-info.outputs.SLACK_COLOR != '' || steps.get-broken-link-info.outputs.SLACK_MESSAGE != '') && github.ref == 'refs/heads/master' }} 
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        if: ${{ steps.get-broken-link-info.outputs.NOTIFY_SLACK == '1' || failure() }}
```

I see that we're notifying not only on errors but also for warnings about broken links.

It seems like we can eliminate the `failure()` condition. If failures cause the step to get skipped without a job status check, maybe we can try `always() && steps.get-broken-link-info.outputs.NOTIFY_SLACK == '1'`. [Docs seem to suggest we might need the job status check](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions).

> If your `if` expression does not contain any of the status functions it will automatically result with `success()`. 

Unless we're intentionally notifying on failure in general, but the Slack message is currently tailored for broken links.

```suggestion
        if: ${{ steps.get-broken-link-info.outputs.NOTIFY_SLACK == '1' }}
```

---

In other words, we don't run the rest of these steps whenever there's any general failure (even unrelated to broken links). But we do want to run them if we've determined that we need to notify Slack, failure or not.

---

I don't think `always()` is the right approach. It will work for all token/authentication portion, but it causes a misleading message when it gets specifically to slack notification.

Currently, the slack bot requires the following `message: "Broken links detected!"` where "Broken links detected" can be anything. The Slack Bot first checks if `blocks` has been passed and if it doesn't it defaults to `message` (otherwise it acts as backup text -- usually in form of pop up what I have seen). 

If there are no warnings or error, it will still fire to the channel that broken links are detected (or any message passed). With failure, it will ensure nothing gets fired, where as always, the slackbot will need to send something (which my thought process is no news is good news in this case). 

Thoughts?

---

Tested workflow and working as intended